tests/cpptests/test_cpp_iterator_not.cpp

```diff
@@ -635,3 +639,173 @@ TEST_F(NotIteratorNoChildTest, Rewind) {
     ASSERT_FALSE(iterator_base->atEOF);
   }
 }
+
+class NotIteratorReducerTest : public ::testing::Test {};
+
+TEST_F(NotIteratorReducerTest, TestNotWithNullChild) {
+  // Test rule 1: If the child is NULL, return a wildcard iterator
+  struct timespec timeout = {LONG_MAX, 999999999};
+  t_docId maxDocId = 100;
+
+  // Create a mock QueryEvalCtx
+  IndexSpec *spec = (IndexSpec*)rm_calloc(1, sizeof(IndexSpec));
+  spec->rule = (SchemaRule*)rm_calloc(1, sizeof(SchemaRule));
+  spec->rule->index_all = false;
+
+  RedisSearchCtx* sctx = (RedisSearchCtx*)rm_calloc(1, sizeof(RedisSearchCtx));
+  sctx->spec = spec;
+
+  DocTable* docTable = (DocTable*)rm_calloc(1, sizeof(DocTable));
+  docTable->maxDocId = maxDocId;
+  docTable->size = maxDocId;
+
+  QueryEvalCtx* qctx = (QueryEvalCtx*)rm_calloc(1, sizeof(QueryEvalCtx));
+  qctx->sctx = sctx;
+  qctx->docTable = docTable;
```

Comment:
Use the existing `MockQueryEvalCtx` instead
```suggestion
  MockQueryEvalCtx mockQctx;
```